pyriemann_qiskit/classification.py

```diff
@@ -743,3 +750,327 @@ def predict(self, X):
+
+    """Nearest Convex Hull Classifier (NCH)
+
+    In NCH, for each class a convex hull is produced by the set of matrices
+    corresponding to each class. There is no training. Calculating a distance
+    to a hull is an optimization problem and it is calculated for each testing
+    sample (SPD matrix) and each hull/class. The minimal distance defines the
+    predicted class.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.0
+
+    Parameters
+    ----------
+    n_jobs : int, (default=6)
+        The number of jobs to use for the computation. This works by computing
+        each of the hulls in parallel.
+    n_hulls_per_class: int, (default 3)
+        The number of hulls used per class.
+    n_samples_per_hull: int, (default 15)
+        Defines how many samples are used to build a hull.
+    hull_type: string, (default "min-hull")
+        Selects how the hull is constructed. Possible values are
+        "min-hull" and "random-hull"
+
+    References
+    ----------
+    .. [1] \
+        K. Zhao, A. Wiliem, S. Chen, and B. C. Lovell,
+        ‘Convex Class Model on Symmetric Positive Definite Manifolds’,
+        Image and Vision Computing, 2019.
+    """
+
+    def __init__(
+        self, n_jobs=6, n_hulls_per_class=3, n_samples_per_hull=10, hull_type="min-hull"
+    ):
+        """Init."""
+        self.n_jobs = n_jobs
+        self.n_samples_per_hull = n_samples_per_hull
+        self.n_hulls_per_class = n_hulls_per_class
+        self.matrices_per_class_ = {}
+        self.debug = False
+        self.hull_type = hull_type
+
+        if hull_type not in ["min-hull", "random-hull"]:
+            raise Exception("Error: Unknown hull type.")
+
+    def fit(self, X, y):
+        """Fit (store the training data).
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y : ndarray, shape (n_matrices,)
+            Labels for each matrix.
+        sample_weight : None
+            Not used, here for compatibility with sklearn API.
+
+        Returns
+        -------
+        self : NearestConvexHull instance
+            The NearestConvexHull instance.
+        """
+
+        if self.debug:
+            print("Start NCH Train")
+        self.classes_ = np.unique(y)
+
+        for c in self.classes_:
+            self.matrices_per_class_[c] = X[y == c]
+
+        if self.debug:
+            print("Samples per class:")
+            for c in self.classes_:
+                print("Class: ", c, " Count: ", self.matrices_per_class_[c].shape[0])
+
+            print("End NCH Train")
+
+    def _process_sample_min_hull(self, test_sample):
+        """Finds the closes N covmats and uses them to build a single hull per class"""
+        distances = []
+
+        for c in self.classes_:
+            distances_to_covs = [
+                distance_logeuclid(test_sample, cov)
+                for cov in self.matrices_per_class_[c]
+            ]
```

Comment:
Using `from pyriemann.utils.distance import distance`, you can simplify code this way
`distances_to_covs = distance(self.matrices_per_class_[c], test_sample, metric="logeuclid")[:, 0]`

Then, you can compute `np.argsort(distances_to_covs)` instead of `np.argsort(np.array(distances_to_covs))`.